### PR TITLE
[orc8r][service_registry] Add GetAnnotationFields method to simplify retrieving strings annotations

### PR DESCRIPTION
**Pull request by hcgatewood**
## Summary

It's turned into a common service registry pattern to retrieve a (comma-separated) list of strings stored as an annotation.

This diff coalesces that functionality into a single method, with a stable and well-documented set of expectations.

Also updated go-funk for some reason, leaving the update in for simplicity.

## Test Plan

- [x] make test
- [x] build everything and ensure an LTE swagger API call succeeds

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: orc8r/lib/go/registry/registry.go [hunk 1]
```diff
@@ -206,12 +207,32 @@ func (r *ServiceRegistry) GetAnnotation(service, annotationName string) (string,
...
+	if err != nil {
+		return nil, err
+	}
+
+	split := strings.Split(annotationValue, fieldSeparator)
+	values := funk.
+		Chain(split).
+		Map(func(s string) string { return strings.Join(strings.Fields(s), "") }).
+		Filter(func(s string) bool { return s != "" }).
+		Value().([]string)
```
Comment (xjtian) [inline orc8r/lib/go/registry/registry.go]: eh at this point it's better to just write a for loop

File: orc8r/lib/go/registry/registry.go [hunk 2]
```diff
@@ -206,12 +207,32 @@ func (r *ServiceRegistry) GetAnnotation(service, annotationName string) (string,
...
 
-	return labelValue, nil
+	return annotationValue, nil
+}
+
+// GetAnnotationFields returns the split fields of the value for the passed
+// annotation name.
+// First splits by field separator, then strips all whitespace
+// (including newlines). Empty fields are removed.
+func (r *ServiceRegistry) GetAnnotationFields(service, annotationName, fieldSeparator string) ([]string, error) {
```
Comment (xjtian) [inline orc8r/lib/go/registry/registry.go]: `GetAnnotationListValue` or `GetListAnnotationValue` or something like that. This name makes me think you're grabbing all the annotation keys on a service. I think you can probably just use the const for the separator too

File: orc8r/lib/go/registry/global_registry.go [hunk 3]
```diff
@@ -120,6 +120,13 @@ func GetAnnotation(service, annotationName string) (string, error) {
 	return globalRegistry.GetAnnotation(service, annotationName)
 }
 
+// GetAnnotationFields returns the split fields of the value for the passed
+// annotation name.
+// The service needs to be added to the registry before this.
+func GetAnnotationFields(service, annotationName, fieldSeparator string) ([]string, error) {
```
Comment (xjtian) [inline orc8r/lib/go/registry/global_registry.go]: bubble up the name change

**Review by xjtian — CHANGES_REQUESTED** [review]

**Review by xjtian — APPROVED** [review]
Rebase and do a `make tidy` to sync up the go.mod/go.sum files before merging

**Review by mpgermano — APPROVED** [review]

**Pull request merged**